samples/gen-js-db/index.d.ts

```diff
@@ -49,129 +49,134 @@ interface AddressOptions {
   address: string;
 }
 
-type SwaggerTestOptions = (DiscoveryOptions | AddressOptions) & GenericOptions; 
+type SwaggerTestOptions = (DiscoveryOptions | AddressOptions) & GenericOptions;
 
+import models = SwaggerTest.Models
 
-type Branch = ("master" | "DEV_BRANCH" | "test");
+declare class SwaggerTest {
+  constructor(options: SwaggerTestOptions);
 
-type Category = ("a" | "b");
+  
+  healthCheck(options?: RequestOptions, cb?: Callback<void>): Promise<void>
+  
+}
 
-type Deployment = {
+declare namespace SwaggerTest {
+  const RetryPolicies: {
+    Single: RetryPolicy;
+    Exponential: RetryPolicy;
+    None: RetryPolicy;
+  }
+
+  const DefaultCircuitOptions: CircuitOptions;
+
+  namespace Errors {
+    
+    class BadRequest {
+  message?: string;
+}
```

Comment:
I'm seeing compilation errors now on code that passes an argument to the constructor of these error types, e.g.
`new ILTokenServiceErrors.NotFound({ message: "stub error" })` now throws an error `Expected 0 arguments, but got 1.`